extensions/liquibase/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/runtime/LiquibaseMongodbClient.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.liquibase.mongodb.runtime;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.inject.Named;
+import jakarta.inject.Qualifier;
+
+/**
+ * Qualifier used to specify which client will be used and therefore which Liquibase MongoDB instance will be injected.
+ * <p>
+ * Liquibase instances can also be qualified by name using @{@link Named}.
+ * The name is the client name prefixed by "liquibase_mongodb_".
+ */
+@Target({ METHOD, FIELD, PARAMETER, TYPE })
+@Retention(RUNTIME)
+@Documented
+@Qualifier
+public @interface LiquibaseMongodbClient {
```

Comment:
Can't we reuse the `@MongoClientName` qualifier?

Reply:
Hi @geoand,

Yes, it’s possible, but I think it could confuse users. They would expect @MongoClientName to resolve a MongoClient, not a LiquibaseMongodbFactory object.

Since the qualifier in this case is actually meant to resolve a LiquibaseMongodbFactory, it might be clearer to rename it, for example, @LiquibaseMongodbFactoryName. This would better reflect its purpose and avoid mixing  with the MongoDB client extension.

What do you think?

Reply:
From the top of my head, I think I would be in favor of using `@MongoClientName` but I will let @geoand decide.